### PR TITLE
crdt transports

### PR DESCRIPTION
**Pull request by gonpombo8**
https://diagrams.menduz.com/#/notebook/XoIWzdc9DQdIMr7XvQy2bvwzInB3/-N1KvXLmQ6IsllIlzOFO

### CODE REVIEW

File: src/engine/types.ts [hunk 1]
```diff
@@ -10,3 +13,23 @@ export type ComponentEcsType<T extends [CompDef, ...CompDef[]]> = {
 export type DeepReadonly<T> = {
   readonly [K in keyof T]: DeepReadonly<T[K]>
 }
+
+/**
+ * @public
+ */
+export type Engine = {
```
Comment (menduz) [inline src/engine/types.ts:20]:
```suggestion
export type IEngine = {
```

File: src/serialization/crdt/componentOperation.ts [hunk 2]
```diff
@@ -23,7 +23,7 @@ export namespace PutComponentOperation {
    *  already allocated
    */
   export function write(
-    entityId: Entity,
+    entity: Entity,
```
Comment (leanmendoza) [inline src/serialization/crdt/componentOperation.ts:26]: like it

File: src/systems/crdt/index.ts [hunk 3]
```diff
@@ -1,132 +1,170 @@
...
+      const buffer = createByteBuffer()
+      for (const message of messagesToProcess) {
+        const { data, timestamp, componentId, entity } = message
+        const crdtMessage: CrdtMessage<Uint8Array> = {
+          key: CrdtUtils.getKey(entity, componentId),
+          data: data,
+          timestamp: timestamp
+        }
+        const component = engine.getComponent(componentId)
+        const currentMessage = crdtClient.processMessage(crdtMessage)
```
Comment (menduz) [inline src/systems/crdt/index.ts:90]: To have a better DX, may this be a boolean signaling a change in state?
```suggestion
        const didChangeState = crdtClient.processMessage(crdtMessage)
```

**Review by leanmendoza — APPROVED** [review]

File: src/systems/crdt/index.ts [hunk 4]
```diff
@@ -1,132 +1,170 @@
...
-      }
-      crdtEntities.set(entity, true)
       for (const componentId of componentsId) {
         const component = engine.getComponent(componentId)
-        const key = CrdtUtils.getKey(entity, componentId)
         const event = crdtClient.createEvent(
-          key,
+          CrdtUtils.getKey(entity, componentId),
           component.toBinary(entity).toBinary()
         )
```
[the remark this answers is not in the document]
Comment (gonpombo8) [reply src/systems/crdt/index.ts:131]: I want the crdt state to have all the information of entities/component. 
So in the future if we compare full states, they have the initial state too.

**Pull request merged**